File: library/src/main/java/com/yy/codex/uikit/UIViewAnimation.java
package com.yy.codex.uikit;

/**
 * Created by cuiminghui on 2017/1/16.
 */

public class UIViewAnimation {

    private boolean mCancelled = false;
    private boolean mFinished = false;

    public boolean isCancelled() {
        return mCancelled;
    }

    public boolean isFinished() {
        return mFinished;
    }

    public void cancel() {
        mCancelled = true;
    }

    public void markFinished() {
        mFinished = true;
    }

}

File: library/src/main/java/com/yy/codex/uikit/UIViewAnimation.kt
package com.yy.codex.uikit

/**
 * Created by cuiminghui on 2017/1/16.
 */

class UIViewAnimation {

    var cancelled = false
        private set
    var finished = false
        private set

    internal var completion: Runnable? = null

    fun cancel() {
        if (!finished) {
            cancelled = true
            completion?.run()
        }
    }

    fun markFinished() {
        finished = true
        completion = null
    }

}
